### PR TITLE
Doc modifications 

### PR DESCRIPTION
**Pull request by animeshsingh**
To make the docs flowing from main readme, as well fixing some syntax/semantic things

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kfserving/69)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,4 +5,4 @@ A KFService encapsulates the complexity of autoscaling, networking, health check
 
 In the future, we hope to support more advanced use cases such as skew detection, explainability, and performance profiling across infrastructure configurations.
 
-This project is an evolution of the original proposal: https://github.com/kubeflow/kubeflow/issues/2306
+This project is an evolution of the [original proposal in the Kubeflow repo](https://github.com/kubeflow/kubeflow/issues/2306). To know more about KFServing architecture, and about the control plan and data plane specifications, please [follow along here](/docs)
```
Comment (ellis-bigelow) [inline README.md]: s/plan/plane

Comment (ellis-bigelow) [reply README.md]: Can we say,

"To learn more about KFServing, [read the docs!](#docs)

File: docs/README.md [hunk 2]
```diff
@@ -1,13 +1,14 @@
 
 # Design Docs
+KFService architecture consists of two main specifications, around control pland and data plane. Control plane is responsible for deploying and managing the lifecycle of models, while data plane is responsible for defining the contracts to interact with models once deployed.
```
Comment (ellis-bigelow) [inline docs/README.md]: KFServing's architecture consists of two main specifications: control plane and data plane.

Comment (ellis-bigelow) [reply docs/README.md]: Your second sentence is somewhat moot with the below bullets.

File: docs/control-plane.md [hunk 3]
```diff
@@ -1,5 +1,7 @@
-# KFServing
-A KFService is a unit of model serving. Users may have many KFServices, including different definitions across dev, test, and prod environments, as well as across different model use cases. KFServices are defined using the [Kubernetes Custom Resources](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) standard. 
+# Control Plane Specification
+
```
Comment (ellis-bigelow) [inline docs/control-plane.md]: nit: remove space

File: docs/control-plane.md [hunk 4]
```diff
@@ -1,5 +1,7 @@
...
+## KFService
+Control plane is responsible for the lifecycle of a KFService. A KFService is a unit of model serving. Users may have many KFServices, including different definitions across dev, test, and prod environments, as well as across different model use cases. KFServices are defined using the [Kubernetes Custom Resources](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) standard.
```
Comment (ellis-bigelow) [inline docs/control-plane.md]: I'd prefer to not have this first sentence.

File: docs/data-plane.md [hunk 5]
```diff
@@ -12,20 +12,20 @@ Its aims:
...
 ### Predictors
 
-Data planes for request/response to machine learning models are the most well defined in the ecosystem. Existing examples are:
+Data planes for request/response to machine learning models are well defined in the ecosystem. Existing examples are:
 
  * [Tensorflow Serving](https://github.com/tensorflow/serving/blob/master/tensorflow_serving/apis/prediction_service.proto)
  * [Seldon Core](https://github.com/SeldonIO/seldon-core/blob/master/proto/prediction.proto)
 
-It is suggested we don't define a new data plane for model input/output at present but allow models to publish the input/output schema they respect.
+At present, it is suggested we don't define a new data plane for model input/output but allow models to publish the input/output schema they respect. In future, we may provide an additional standard data plane schema for kfserving, independent of the backing model runtimes.
```
Comment (ellis-bigelow) [inline docs/data-plane.md]: I might drop "it is suggested"

**Comment by ellis-bigelow** [comment]
Thank you for your contributions!!

File: docs/README.md (hunk 2 above)
Comment (animeshsingh) [reply docs/README.md]: thanks - intentional though. The top level is trying to lay it our in more general sense, or is a language which is more widely understood. 'Managing the lifecycle of deployed models' reaches out to everyone (architects, PMs etc.) , as opposed to 'Defines the specification of the `kfserving` resource'

File: README.md (hunk 1 above)
Comment (animeshsingh) [reply README.md]: Fixed

File: docs/control-plane.md (hunk 3 above)
Comment (animeshsingh) [reply docs/control-plane.md]: Done

File: docs/control-plane.md (hunk 4 above)
Comment (animeshsingh) [reply docs/control-plane.md]: Done

File: docs/data-plane.md (hunk 5 above)
Comment (animeshsingh) [reply docs/data-plane.md]: Done

File: docs/README.md (hunk 2 above)
Comment (ellis-bigelow) [reply docs/README.md]: s/KFService/KFServing

Comment (animeshsingh) [reply docs/README.md]: replace everything with KFService to KFServing @ellis-bigelow ?

Comment (ellis-bigelow) [reply docs/README.md]: KFServing refers to the overall project, KFService refers to the specific resource. I was referring to just this instance.

Comment (animeshsingh) [reply docs/README.md]: Done

**Comment by animeshsingh** [comment]
All changes accounted for @ellis-bigelow

**Comment by ellis-bigelow** [comment]
So close. Missed a couple.

File: docs/README.md [hunk 6]
```diff
@@ -1,13 +1,14 @@
 
 # Design Docs
+KFServing architecture consists of two main specifications, around control pland and data plane. Control plane is responsible for deploying and managing the lifecycle of models, while data plane is responsible for defining the contracts to interact with models once deployed.
```
Comment (ellis-bigelow) [inline docs/README.md]: KFServing's
pland -> plane

Comment (animeshsingh) [reply docs/README.md]: ah - should do a spell check next time :-)

**Pull request merged**